### PR TITLE
chore(tests): migrate prometheus integration tests

### PR DESCRIPTION
**Pull request by jdrouet**
Following #10466, this PR moves the prometheus tests to use docker-compose on the CI.

Signed-off-by: Jérémie Drouet <jeremie.drouet@datadoghq.com>

### CODE REVIEW

File: scripts/integration/docker-compose.prometheus.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+services:
+  influxdb:
```
Comment (spencergilbert) [inline scripts/integration/docker-compose.prometheus.yml:2]: 🤔 why do we start influxdb for prometheus tests?

**Review by spencergilbert — APPROVED** [review]

File: scripts/integration/docker-compose.prometheus.yml (hunk 1 above)
Comment (jdrouet) [reply scripts/integration/docker-compose.prometheus.yml:2]: It's required for the `remote_write` sink [here](https://github.com/vectordotdev/vector/blob/master/src/sinks/prometheus/remote_write.rs#L470)

Comment (spencergilbert) [reply scripts/integration/docker-compose.prometheus.yml:2]: Seems weird/unnecessary 🤷

Comment (bruceg) [reply scripts/integration/docker-compose.prometheus.yml:2]: The prometheus server doesn't actually implement the remote_write protocol, while influxdb does, so I used it to validate the protocol compliance against something other than our own source.

Comment (spencergilbert) [reply scripts/integration/docker-compose.prometheus.yml:2]: it does as of today, `docker run --rm -it prom/prometheus --enable-feature=remote-write-receiver --config.file=/etc/prometheus.yml^C`

Comment (bruceg) [reply scripts/integration/docker-compose.prometheus.yml:2]: Ah, right, I'm forgetting my history again. When I wrote the remote-write setup, I had not found that, and the prometheus documentation seemed (to me at least) to point away from itself. So at this point it is a historical artifact that should probably be rectified.

Comment (spencergilbert) [reply scripts/integration/docker-compose.prometheus.yml:2]: I mean, to be fair, it seems pretty undocumented and unclear how stable/supported that is today 🤷

Comment (jdrouet) [reply scripts/integration/docker-compose.prometheus.yml:2]: @spencergilbert or @bruceg, do you mind if we open an issue to tackle that in a latter PR and we keep this one as it is?

Comment (bruceg) [reply scripts/integration/docker-compose.prometheus.yml:2]: Absolutely that belongs in a separate issue. It's completely unrelated to this change.

**Pull request merged**